sdk/identity/azure-identity/src/main/java/com/azure/identity/ArcIdentityCredential.java

```diff
@@ -49,6 +49,6 @@ public Mono<AccessToken> authenticate(TokenRequestContext request) {
                     + "with the system assigned identity omit the client id when constructing the"
                     + " ManagedIdentityCredential.", null)));
         }
-        return identityClient.authenticateWithManagedIdentityConfidentialClient(request);
+        return identityClient.authenticateWithManagedIdentityMsalClient(request);
```

Comment:
consider: adding a quirks mode to fall-back to the in-house MI implementation in case of support incidents (to unblock customers), which can later be removed after we have full confidence in MSAL's new MI support.